plugin/impl.go

```diff
@@ -9,107 +15,123 @@ import (
+	if len(p.settings.server) == 0 {
+		p.settings.server = getServerWithDefaults("", p.pipeline.System.Host, p.pipeline.System.Proto)
+	}
+	if len(p.settings.server) == 0 {
+		return fmt.Errorf("you must provide your drone server")
 	}
 
-	if p.Wait && p.LastSuccessful {
-		return fmt.Errorf("Error: only one of wait and last_successful can be true; choose one")
+	if p.settings.Wait && p.settings.LastSuccessful {
+		return fmt.Errorf("only one of wait and last_successful can be true; choose one")
 	}
 
-	params, err := parseParams(p.Params)
+	var err error
+	p.settings.params, err = parseParams(p.settings.Params.Value())
 	if err != nil {
-		return fmt.Errorf("Error: unable to parse params: %s.\n", err)
+		return fmt.Errorf("unable to parse params: %s", err)
 	}
 
-	for _, k := range p.ParamsEnv {
+	for _, k := range p.settings.ParamsEnv.Value() {
 		v, exists := os.LookupEnv(k)
 		if !exists {
-			return fmt.Errorf("Error: param_from_env %s is not set.\n", k)
+			return fmt.Errorf("param_from_env %s is not set", k)
 		}
 
-		params[k] = v
+		p.settings.params[k] = v
 	}
 
+	return nil
+}
+
+// Execute provides the implementation of the plugin.
+func (p *Plugin) Execute() error {
 	config := new(oauth2.Config)
 
 	auther := config.Client(
-		oauth2.NoContext,
+		context.WithValue(context.Background(), oauth2.HTTPClient, p.network.Client),
 		&oauth2.Token{
-			AccessToken: p.Token,
+			AccessToken: p.settings.Token,
 		},
 	)
 
-	client := drone.NewClient(p.Server, auther)
+	client := drone.NewClient(p.settings.server, auther)
 
-	for _, entry := range p.Repos {
+	for _, entry := range p.settings.Repos.Value() {
 
 		// parses the repository name in owner/name@branch format
 		owner, name, branch := parseRepoBranch(entry)
 		if len(owner) == 0 || len(name) == 0 {
-			return fmt.Errorf("Error: unable to parse repository name %s.\n", entry)
+			return fmt.Errorf("unable to parse repository name %s", entry)
 		}
 
 		// check for mandatory build no during deploy trigger
-		if len(p.Deploy) != 0 {
+		if len(p.settings.Deploy) != 0 {
 			if branch == "" {
-				return fmt.Errorf("Error: build no or branch must be mentioned for deploy, format repository@build/branch")
+				return fmt.Errorf("build no or branch must be mentioned for deploy, format repository@build/branch")
 			}
-			if _, err := strconv.Atoi(branch); err != nil && !p.LastSuccessful {
-				return fmt.Errorf("Error: for deploy build no must be numeric only " +
+			if _, err := strconv.Atoi(branch); err != nil && !p.settings.LastSuccessful {
+				return fmt.Errorf("for deploy build no must be numeric only " +
 					" or for branch deploy last_successful should be true," +
 					" format repository@build/branch")
 			}
 		}
 
 		waiting := false
 
-		timeout := time.After(p.Timeout)
+		timeout := time.After(p.settings.Timeout)
+		//lint:ignore SA1015 refactor later
 		tick := time.Tick(1 * time.Second)
 
+		var err error
```

Comment:
Did I miss something or is this not required?

Reply:
Originally the plugin had an `Execute` that had all the logic. When the validation was split into `Validate` the `Execute` bits did not have `err` declared. So yea its necessary.